scripts/py_matter_yamltests/matter_yamltests/parser.py

```diff
@@ -481,6 +492,52 @@ def post_process_response(self, response: dict):
 
         return result
 
+    def _response_cluster_wait_validation(self, response, result):
+        """Check if the response concrete path match the configuration of the test step
+           and validate that the response type (e.g readAttribute/writeAttribute/...) match
+           the expectation from the test step."""
+        check_type = PostProcessCheckType.WAIT_VALIDATION
+        error_success = 'The test expectation "{wait_for}" for "{cluster}.{wait_type}" on endpoint {endpoint} is true'
+        error_failure = 'The test expectation "{expected} == {received}" is false'
+
+        if self.is_attribute:
+            expected_wait_type = self.attribute
+            received_wait_type = response.get('attribute')
+        elif self.is_event:
+            expected_wait_type = self.event
+            received_wait_type = response.get('event')
+        else:
+            expected_wait_type = self.command
+            received_wait_type = response.get('command')
+
+        expected_values = [
+            self.wait_for,
+            self.endpoint,
+            # The name in tests does not always use spaces
```

Comment:
We should file a followup to just fix that...

Reply:
I have opened https://github.com/project-chip/connectedhomeip/issues/24619